src/new-components/box/margin.module.css

```diff
@@ -0,0 +1,507 @@
+/***************/
+/* margin-top */
+/***************/
+
+.marginTop-xsmall {
+    margin-top: var(--reactist-spacing-xsmall);
+}
+.marginTop-small {
+    margin-top: var(--reactist-spacing-small);
+}
+.marginTop-medium {
+    margin-top: var(--reactist-spacing-medium);
+}
+.marginTop-large {
+    margin-top: var(--reactist-spacing-large);
+}
+.marginTop-xlarge {
+    margin-top: var(--reactist-spacing-xlarge);
+}
+.marginTop-xxlarge {
+    margin-top: var(--reactist-spacing-xxlarge);
+}
+
+/* negative margin-top */
+.marginTop--xsmall {
+    margin-top: calc(var(--reactist-spacing-xsmall) * -1);
+}
+.marginTop--small {
+    margin-top: calc(var(--reactist-spacing-small) * -1);
+}
+.marginTop--medium {
+    margin-top: calc(var(--reactist-spacing-medium) * -1);
+}
+.marginTop--large {
+    margin-top: calc(var(--reactist-spacing-large) * -1);
+}
+.marginTop--xlarge {
+    margin-top: calc(var(--reactist-spacing-xlarge) * -1);
+}
+.marginTop--xxlarge {
+    margin-top: calc(var(--reactist-spacing-xxlarge) * -1);
+}
+
+/* margin-top for tablet */
+@media (min-width: 768px /* --reactist-breakpoint-tablet */) {
+    .tablet-marginTop-xsmall {
```

Comment:
I dug into how these classes are getting applied based on the values provided to the props, it's really cool!